server/src/main/java/org/opensearch/action/admin/cluster/awarenesshealth/ClusterAwarenessHealthResponse.java

```diff
@@ -0,0 +1,240 @@
+    }
+
+    public int getUnassignedShards() {
+        return clusterStateHealth.getUnassignedShards();
+    }
+
+    public int getInitializingShards() {
+        return clusterStateHealth.getInitializingShards();
+    }
+
+    public int getRelocatingShards() {
+        return clusterStateHealth.getRelocatingShards();
+    }
+
+    public int getNumberOfDataNodes() {
+        return clusterStateHealth.getNumberOfDataNodes();
+    }
+
+    public int getNumberOfNodes() {
+        return clusterStateHealth.getNumberOfNodes();
+    }
+
+    public int getActivePrimaryShards() {
+        return clusterStateHealth.getActivePrimaryShards();
+    }
+
+    public double getActiveShardsPercent() {
+        return clusterStateHealth.getActiveShardsPercent();
+    }
+
+    public ClusterHealthStatus getStatus() {
+        return clusterStateHealth.getStatus();
+    }
+
+    public boolean hasDiscoveredClusterManager() {
+        return clusterStateHealth.hasDiscoveredClusterManager();
+    }
+
+    public int getNumberOfPendingTasks() {
+        return numberOfPendingTasks;
+    }
+
+    public TimeValue getMaxWaitTime() {
+        return maxWaitTime;
+    }
+
+    public int getNumberOfInflightFetches() {
+        return numberOfInflightFetches;
+    }
+
+    public ClusterAwarenessAttributesHealth getAwarenessAttributeHealth() {
+        return clusterAwarenessHealth.getAwarenessAttributeHealth();
+    }
+
+    public int getDelayedUnassignedShards() {
+        return delayedUnassignedShards;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(CLUSTER_NAME, getClusterName());
+        builder.field(STATUS, getStatus().name().toLowerCase(Locale.ROOT));
+        builder.field(NUMBER_OF_NODES, getNumberOfNodes());
+        builder.field(NUMBER_OF_DATA_NODES, getNumberOfDataNodes());
+        builder.field(DISCOVERED_CLUSTER_MANAGER, hasDiscoveredClusterManager());
+        builder.field(ACTIVE_PRIMARY_SHARDS, getActivePrimaryShards());
+        builder.field(ACTIVE_SHARDS, getActiveShards());
+        builder.field(RELOCATING_SHARDS, getRelocatingShards());
+        builder.field(INITIALIZING_SHARDS, getInitializingShards());
+        builder.field(UNASSIGNED_SHARDS, getUnassignedShards());
+        builder.field(DELAYED_UNASSIGNED_SHARDS, getDelayedUnassignedShards());
+        builder.field(NUMBER_OF_PENDING_TASKS, getNumberOfPendingTasks());
+        builder.field(NUMBER_OF_IN_FLIGHT_FETCH, getNumberOfInflightFetches());
+        builder.humanReadableField(TASK_MAX_WAIT_TIME_IN_QUEUE_IN_MILLIS, TASK_MAX_WAIT_TIME_IN_QUEUE, getMaxWaitTime());
+        builder.percentageField(ACTIVE_SHARDS_PERCENT_AS_NUMBER, ACTIVE_SHARDS_PERCENT, getActiveShardsPercent());
+        getAwarenessAttributeHealth().toXContent(builder, params);
+        builder.endObject();
+        return builder;
+    }
```

Comment:
Corresponding `fromXContent`?

Reply:
That is a very large function and major is tested on service layer itself.